compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -804,6 +827,51 @@ object JavaParsers {
       addCompanionObject(statics, cls)
     }
 
+    def recordDecl(start: Offset, mods: Modifiers): List[Tree] =
+      accept(RECORD)
+      val nameOffset = in.offset
+      val name = identForType()
+      val tparams = typeParams()
+      val header = formalParams()
+      val superclass = javaLangRecord() // records always extend java.lang.Record
+      val interfaces = interfacesOpt() // records may implement interfaces
+      val (statics, body) = typeBody(RECORD, name, tparams)
+
+      // We need to generate accessors for every param, if no method with the same name is already defined
+
+      var fieldsByName = header.map(v => (v.name, (v.tpt, v.mods.annotations))).to(ListMap)
```

Comment:
Any specific reason for using a ListMap instead of a LinkedHashMap? Since ListMap#contains takes an amount of time linear in the number of entries in the map and since we're calling it on every parameterless method in the class in the loop below, our run-time will be quadratic in the number of parameterless methods in the class which could start mattering if someone has a class with 100 methods like that. Using a LinkedHashMap would avoid having to worry about this.

Reply:
IIRC the reason is that I wrote `import scala.collection.immutable.` and `LinkedHashMap` didn't show up because it only exists in its mutable variant. But I agree that it would be a better fit.

Reply:
replaced by LinkedHashMap :)